tests/v1beta3/authconfig.yaml

```diff
@@ -130,27 +115,21 @@ spec:
           }
     admin-kubernetes-rbac:
       when:
-      - patternRef: admin-path
-      - selector: auth.identity.kubernetes-rbac
-        operator: eq
-        value: 'true'
+      - predicate: request.http.path.matches("^/admin(/.*)?$")
+      - predicate: auth.identity["kubernetes-rbac"]
```

Comment:
It looks like this predicate is failing.

Reply:
Well... _may_ be failing. Or at least flaky.

Reply:
But ... then succeeds?

Reply:
Hmm. Maybe not. The e2e test that failed was a GET request to `/greetings/1` authenticating by K8s SA token.

Could this other predicate be the one to blame `request.http.path.matches("^/admin(/.*)?$")`? 🤔

Reply:
Could it be that it failed to [obtain a token from the cluster](https://github.com/Kuadrant/authorino/blob/1bea62698b1e9a3485a87f17106cece0ee1ac03f/tests/e2e-test.sh#L119) by setting `$access_token` to some non-empty value that nonetheless is not a valid token, then missed all authentication methods, defaulting to `anonymous` (`priority`: 1) and thus skipping the authorisation rules that otherwise would be enforced when `auth.identity["kubernetes-rbac"]`?

I'm now inclined to blame on the e2e tests themselves, I guess. It's badly written. It shouldn't accept _any_ string as a valid access token ready to send the request.